accounts/abi/bind/bind.go

```diff
@@ -162,10 +162,13 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 			return "", errors.New("java binding for tuple arguments is not supported yet")
 		}
 
+		inputBin := strings.TrimPrefix(strings.TrimSpace(bytecodes[i]), "0x")
+		outputBin := strings.SplitAfter(inputBin, "6080604052")
 		contracts[types[i]] = &tmplContract{
 			Type:        capitalise(types[i]),
 			InputABI:    strings.Replace(strippedABI, "\"", "\\\"", -1),
-			InputBin:    strings.TrimPrefix(strings.TrimSpace(bytecodes[i]), "0x"),
+			InputBin:    inputBin,
+			OutputBin:   outputBin[len(outputBin)-1],
```

Comment:
Did you mean `outputBin[:len(outputBin)]` ?

Reply:
I don't think so, my intuition was that you want the last element of `outputBin` to be `OutputBin`
I'm generally not very committed to this change, we should maybe discuss in stabby

Reply:
How about storing the hash of the output binary? This way we don't have to compare _all_ the bytecode and it will quickly determine if the code is the one we are looking for.